### PR TITLE
fix pytroch on yarn model output is NaN

### PR DESCRIPTION
**Pull request by qiuxin2012**
ModelBroadcast will set weights to empty tensor(not null).
When clone model in executor, this unnecessary updateWeightNative will lead to out of bound access as this weights is an empty tensor. Just remove it to fix the NaN problem.

### CODE REVIEW

**Comment by jason-dai** [comment]
@hhbyyh please take a look.

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TorchNet.scala [hunk 1]
```diff
@@ -58,8 +58,6 @@ class TorchNet private(private val modelHolder: TorchModelHolder)
     if (weights == null) {
       val w = PytorchModel.getWeightNative(ref).clone()
       weights = Tensor(w, Array(w.length))
-    } else {
-      PytorchModel.updateWeightNative(ref, weights.storage().array())
```
Comment (hhbyyh) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TorchNet.scala]: I'm not sure if we should remove it entirely. For inference, we still need to update the weights in native model.

**Comment by qiuxin2012** [comment]
find another issue, validation accuracy doesn't match training accuracy.
```
2019-09-19 09:01:56 INFO  DistriOptimizer$:406 - [Epoch 2 1984/1992][Iteration 497][Wall Clock 633.823767662s] Trained 8 records in 1.265377379 seconds. Throughput is 6.3222246 records/second. Loss is 0.026811982. TorchNetb7f9f0b8's hyper parameters: Current learning rate is 0.001. Current dampening is 1.7976931348623157E308.  
update weight native training
update weight native training
2019-09-19 09:01:58 INFO  DistriOptimizer$:406 - [Epoch 2 1992/1992][Iteration 498][Wall Clock 635.090751486s] Trained 8 records in 1.266983824 seconds. Throughput is 6.314208 records/second. Loss is 4.2548898E-4. TorchNetb7f9f0b8's hyper parameters: Current learning rate is 0.001. Current dampening is 1.7976931348623157E308.  
2019-09-19 09:01:58 INFO  DistriOptimizer$:451 - [Epoch 2 1992/1992][Iteration 498][Wall Clock 635.090751486s] Epoch finished. Wall clock time is 640696.282168 ms
2019-09-19 09:01:58 INFO  DistriOptimizer$:111 - [Epoch 2 1992/1992][Iteration 498][Wall Clock 635.090751486s] Validate model...
2019-09-19 09:02:03 INFO  DistriOptimizer$:177 - [Epoch 2 1992/1992][Iteration 498][Wall Clock 635.090751486s] validate model throughput is 42.208885 records/second
2019-09-19 09:02:03 INFO  DistriOptimizer$:180 - [Epoch 2 1992/1992][Iteration 498][Wall Clock 635.090751486s] Top1Accuracy is Accuracy(correct: 220, count: 230, accuracy: 0.9565217391304348)
creating: createToTuple
creating: createChainedPreprocessing
TorchNet loading in TorchNet[b7f9f0b8]
update weight native
-0.0101034595
-0.0059452774
-0.0016931365
...
0.016581625
0.017425977
0.0254069
[com.intel.analytics.bigdl.tensor.DenseTensor$mcF$sp of size 11691514]
[Stage 2016:>                                                       (0 + 1) / 1]TorchNet loading in TorchNet[b7f9f0b8]
update weight native
Empty Tensor
+--------------------+------------+-----+----------+                            
|               image|        name|label|prediction|
+--------------------+------------+-----+----------+
|[hdfs://localhost...|  cat.71.jpg|  1.0|       1.0|
|[hdfs://localhost...|cat.7237.jpg|  1.0|       1.0|
|[hdfs://localhost...| cat.757.jpg|  1.0|       1.0|
|[hdfs://localhost...|cat.7747.jpg|  1.0|       1.0|
|[hdfs://localhost...|  cat.79.jpg|  1.0|       1.0|
|[hdfs://localhost...|cat.7926.jpg|  1.0|       1.0|
|[hdfs://localhost...|cat.7987.jpg|  1.0|       1.0|
|[hdfs://localhost...|dog.7006.jpg|  0.0|       1.0|
|[hdfs://localhost...|dog.7014.jpg|  0.0|       0.0|
|[hdfs://localhost...|dog.7147.jpg|  0.0|       1.0|
|[hdfs://localhost...|dog.7188.jpg|  0.0|       1.0|
|[hdfs://localhost...|dog.7211.jpg|  0.0|       1.0|
|[hdfs://localhost...| dog.722.jpg|  0.0|       1.0|
|[hdfs://localhost...|dog.7241.jpg|  0.0|       1.0|
|[hdfs://localhost...|dog.7307.jpg|  0.0|       1.0|
|[hdfs://localhost...| dog.738.jpg|  0.0|       1.0|
|[hdfs://localhost...|dog.7557.jpg|  0.0|       1.0|
|[hdfs://localhost...|dog.7591.jpg|  0.0|       1.0|
|[hdfs://localhost...|dog.7628.jpg|  0.0|       1.0|
|[hdfs://localhost...|dog.7745.jpg|  0.0|       1.0|
+--------------------+------------+-----+----------+
only showing top 20 rows

Validation accuracy = 0.508696 

```

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TorchNet.scala (hunk 1 above)
Comment (qiuxin2012) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TorchNet.scala]: Remove entirely will cause another error. So add a condition to walk around the out-of-bound access.

**Comment by qiuxin2012** [comment]
jenkins passed

**Comment by hhbyyh** [comment]
LGTM

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TorchNet.scala (hunk 1 above)
Comment (jason-dai) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TorchNet.scala]: use `!weights.empty()` instead

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/nnframes/NNEstimator.scala [hunk 2]
```diff
@@ -654,6 +654,7 @@ class NNModel[T: ClassTag] private[zoo] (
     // concat the prediction and other columns in DF. avoid zip between RDD
     val resultRDD = dataFrame.rdd.mapPartitions { rowIter =>
       val localModel = modelBroadCast.value()
+      localModel.evaluate()
```
Comment (qiuxin2012) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/nnframes/NNEstimator.scala]: @jason-dai The Accucary mismatching is because this localModel's native weight doesn't update to the latest weight, it's still the origin weight loading from pytorch. So I override `evaluate()` method to update latest weight to native, and call evaluate() in mapPartitions here.
@hhbyyh is agree with this solution, do you have any suggestion about this?

**Comment by qiuxin2012** [comment]
jenkins passed
